core/src/main/java/io/aiven/klaw/config/MigrationUtility.java

```diff
@@ -29,64 +35,98 @@
 public class MigrationUtility {
 
   public static final int SUPPORTED_KLAW_VERSION_NUMBER_SYSTEM = 3;
+  public static final String QUERY_DATEEXECUTED_FROM_DATABASECHANGELOG =
+      "SELECT DATEEXECUTED FROM %s";
+  public static final int ALLOWED_TIME_BETWEEN_DB_UPDATES_TO_BE_FIRST_INSTALL = 1;
+  public static final String DATA_VERSION_DEFAULT = "0.0.0";
 
   @Value("${klaw.data.migration.packageToScan:io.aiven.klaw.dao.migration}")
   private String packageToScan;
 
   @Value("${klaw.version}")
   private String currentKlawVersion;
 
+  @Value("${klaw.db.hoursFor.tableCreation:1}")
+  private Integer allowedTimeBetweenTableInstall;
+
   @Autowired private DataVersionRepo versionRepo;
 
   @Autowired private ApplicationContext context;
 
+  @Autowired private SpringLiquibase liquibase;
+
+  private boolean isCleanInstall;
+
   @SchedulerLock(
       name = "TaskScheduler_MigrationUtility",
       lockAtLeastFor = "${klaw.shedlock.lockAtLeastFor:PT30M}",
       lockAtMostFor = "${klaw.shedlock.lockAtMostFor:PT60M}")
   public void startMigration() throws Exception {
+
     // Find the latest version in DB
-    String latestDataVersion = getLatestDataVersion();
-    if (!isVersionGreaterThenCurrentVersion(latestDataVersion, currentKlawVersion)) {
+    DataVersion currentDataVersion = getLatestDataVersion();
+
+    String latestDataVersion = getLatestDataVersionOrDefault(currentDataVersion);
+    if (latestDataVersion.equals(DATA_VERSION_DEFAULT) && isNewInstall()) {
+      log.info(
+          "This is a new install and no data migration is required. Setting Klaw Version to {}",
+          currentKlawVersion);
+      updateDataVersionInDB(currentKlawVersion, 0);
+      return;
+    } else if (!isVersionGreaterThenCurrentVersion(latestDataVersion, currentKlawVersion)) {
       log.info(
           "Current Data Version {} is greater than the Klaw version {}, no action needed.",
           latestDataVersion,
           currentKlawVersion);
       return;
+      // This else if is only called when there is a new install of Klaw or an update from a version
+      // pre Migration Utility.
     }
 
```

Comment:
As an example : my local branch is now on 2.5.0. Still it executes the migration in 2.5.0. It shouldn't imo